src/elements/content-sidebar/activity-feed/comment-form/CommentForm.js

```diff
@@ -129,7 +134,7 @@ class CommentForm extends React.Component<Props, State> {
             <Media className={inputContainerClassNames}>
                 {!isEditing && (
                     <Media.Figure className="bcs-CommentForm-avatar">
-                        <Avatar getAvatarUrl={getAvatarUrl} user={user} />
+                        {renderAvatar(<Avatar getAvatarUrl={getAvatarUrl} user={user} />)}
```

Comment:
Should this be renamed to something like renderAvatarWrapper? or renderAvatarTooltip?

Reply:
> renderAvatarTooltip?

I consciously avoided very specific name that would connect this property with rendering a tooltip. That is because potentially a property like this could be reused in other contexts; those contexts could add something totally different than the tooltip.

> renderAvatarWrapper

I see value in `renderAvatarWrapper` vs `renderAvatar` - it clearly expresses we are wrapping the avatar with something. One problem I see with that particular name is that Wrapper components refer only to the outside component but not the inside component, so `renderAvatarWrapper` might suggest rendering the Wrapper but not the Avatar itself. `renderAvatarWithWrapper` might be better. Or `wrapAvatar`, but that one drops `render` prefix, which connects to [Render Props](https://reactjs.org/docs/render-props.html) pattern which this exactly is.

Reply:
I think `renderAvatarWithWrapper` makes sense.

Reply:
I hate to make it more verbose, but would it make sense to have it be `renderCommentFormAvatar` to be more descriptive of what it does? Initially upon reading the prop I thought it was for all avatars in the activity feed.

Reply:
>

> I hate to make it more verbose, but would it make sense to have it be `renderCommentFormAvatar` to be more descriptive of what it does? Initially upon reading the prop I thought it was for all avatars in the activity feed.

That definitely makes sense. I didn't think there might be more avatar occurrences than just in `CommentForm`.

For now I refactored the name to `renderCommentFormAvatar`. I'm discussing with @ConradJChan how this function should work, which will affect the other changes to the name. In the current form I guess `WithWrapper` isn't the best, as `renderAvatar`/`renderCommentFormAvatar` might wrap the avatar in something or it might now - it's up to the parent component to decide what they do with the avatar; they might even replace it with some totally different implementation of avatar.